transport/grpc/outbound.go

```diff
@@ -107,7 +107,14 @@ func (o *Outbound) Call(ctx context.Context, request *transport.Request) (*trans
 
 	var responseBody []byte
 	var responseMD metadata.MD
-	invokeErr := o.invoke(ctx, request, &responseBody, &responseMD, start)
+	span, invokeErr := o.invoke(ctx, request, &responseBody, &responseMD, start)
```

Comment:
This code may require a few more changes.

Right now you're not distinguishing between an `invoke` error or a service match error. The invoke call is a fundamental layer below the service matching. Therefore, we should return that error (if any) before returning the service match error (what this change is doing now).

If we have an `invokeErr`, there's not much point checking the service name, since, for instance, the entire request may have failed.

Reply:
The return statement in this code looks a little too clever for my liking. Feel free to change it so we don't return the `transport.Request` _and_ an error if there is one.

Reply:
Resolved. Return order and interface have been changed and simplified.